diff --git a/leaderboard/leaderboard/utils/route_parser.py b/leaderboard/leaderboard/utils/route_parser.py
--- a/leaderboard/leaderboard/utils/route_parser.py
+++ b/leaderboard/leaderboard/utils/route_parser.py
@@ -129,7 +129,7 @@ class RouteParser(object):
                 scenario_config = ScenarioConfiguration()
                 scenario_config.type = scenario.attrib.get('type', None)
                 scenario_config.name = scenario.attrib.get('name', None)
-                for elem in scenario.getchildren():
+                for elem in list(scenario):
                     setattr(scenario_config, elem.tag, elem.attrib)
 
                 config.scenario_configs.append(scenario_config)
diff --git a/scenario_runner/srunner/tools/scenario_parser.py b/scenario_runner/srunner/tools/scenario_parser.py
--- a/scenario_runner/srunner/tools/scenario_parser.py
+++ b/scenario_runner/srunner/tools/scenario_parser.py
@@ -68,7 +68,7 @@ class ScenarioConfigurationParser(object):
                 config.town = town
                 config.name = scenario.attrib.get('name', None)
                 config.type = scenario.attrib.get('type', None)
-                for elem in scenario.getchildren():
+                for elem in list(scenario):
                     if elem.tag == 'ego_vehicle':
                         config.ego_vehicles.append(ActorConfigurationData.parse_from_node(elem, 'hero'))
                     elif elem.tag == 'other_actor':
diff --git a/scenario_runner/srunner/tools/route_parser.py b/scenario_runner/srunner/tools/route_parser.py
--- a/scenario_runner/srunner/tools/route_parser.py
+++ b/scenario_runner/srunner/tools/route_parser.py
@@ -80,7 +80,7 @@ class RouteParser(object):
                 scenario_config = ScenarioConfiguration()
                 scenario_config.type = scenario.attrib.get('type', None)
                 scenario_config.name = scenario.attrib.get('name', None)
-                for elem in scenario.getchildren():
+                for elem in list(scenario):
                     setattr(scenario_config, elem.tag, elem.attrib)
 
                 config.scenario_configs.append(scenario_config)
